kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/CipherCode.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.encryption;
+
+/**
+ * <p>Identifies a particular cipher, and may include values for some or all of its parameters.</p>
+ *
+ * <p>The ciphers available to be used depends on the {@link EncryptionVersion}:
+ * You cannot add a new cipher to an existing EncryptionVersion because doing
+ * so would expose old versions of the software to ciphers that they don't know
+ * about meaning they won't be able to supprt the decryption guarantee.</p>
+ */
+public enum CipherCode {
```

Comment:
I'm keen that _eventually_ (post 0.5.0 timeframe) we add support for at least one other cipher. XChaCha20Poly1305 is one candidate, though it's not available in JCE currently.